src/Designer/frontend/libs/studio-components/src/components/StudioToggleableTextfield/StudioToggleableTextfield.tsx

```diff
@@ -34,15 +34,20 @@ function StudioToggleableTextfield(
     defaultValue,
     ...rest
   }: StudioToggleableTextfieldProps,
-  ref: Ref<HTMLInputElement>,
+  ref: Ref<HTMLDivElement>,
 ): ReactElement {
   const [isViewMode, setIsViewMode] = useState<boolean>(true);
   const [errorMessage, setErrorMessage] = useState<string | undefined>(undefined);
+  const [currentValue, setCurrentValue] = useState<string>(String(value ?? defaultValue ?? ''));
 
   useEffect(() => {
     if (onIsViewMode) onIsViewMode(isViewMode);
   }, [isViewMode, onIsViewMode]);
 
+  useEffect(() => {
+    setCurrentValue(String(value ?? defaultValue ?? ''));
+  }, [value, defaultValue]);
+
```

Comment:
Is it possible to remove this useEffect? 
It seems like we could use `currentValue` in the StudioIconTextField´s value prop, and then update currentValue using setCurrentValue in the changeHandler like it already does.